common/scripts/next-beta-version.js

```diff
@@ -0,0 +1,21 @@
+const semver = require('semver');
+
+function nextBetaVersion(currentVersion, minor) {
+  let nextBetaVersion = currentVersion;
+
+  if (minor) {
+    nextBetaVersion = semver.inc(currentVersion, 'preminor', 'beta', '1');
+  } else {
+    nextBetaVersion = semver.inc(currentVersion, 'prepatch', 'beta', '1');
+  }
+
+  const oldBetaBuild = Number.parseInt(currentVersion.split('-beta.')[1]);
```

Comment:
I think we should do this check first no?

Reply:
This part of the code is to increment the beta build number (X in 1.0.0-beta.X) of the current version and use for the next beta build version. I added code comment and renamed the variables to be more clear.